uaclient/cli.py

```diff
@@ -922,6 +922,7 @@ def _attach_with_token(
         print(ua_status.MESSAGE_ATTACH_SUCCESS_NO_CONTRACT_NAME)
 
     config.update_ua_messages(cfg)
+    config.write_cache("marker-attached", "")
```

Comment:
If we are going through that route, do we want to bring the service back if the user runs a `detach` operation ? We could remove the `marker-attached`  file on the `action_detach` function, but we would need to manually restart systemd as well, right ?

Reply:
detach is calling `delete_cache` which will delete the `marker-attached` file, but yes I haven't determined if we'll need to prod systemd to notice that yet. We'll see after I change approaches to using a `.path` instead of a ConditionPathExists

Reply:
this is now updated so that attach calls an `disable_license_check` function and detach calls an `enable_license_check` function